### DISCUSSION TITLE
How to use Mod-Tap with layer toggle

### DISCUSSION THREAD

**Discussion by monandszy** [vial-kb/vial-gui#258] [Q&A]
I tried, but it doesn't work, also layers don't show in the gui, so I assume It's not the correct way.
LCTL_T((TG(2))
LALT_T((DF(1))
LALT_T(DF(0))
I tried tap dance, but the delay never feels right. I just want to achieve a simple behaviour:
On key release, if it's the only key that was pressed, toggle the layer immediately.
On hold When another key is pressed act as control, just like mod tap.
No delays, forced waiting etc. I use this on a daily basis, and It works perfectly LSFT_T(KC_SPACE).
I've only used the ui so far, is there a way to modify the tap dance to acheive this behaviour via code?

**Comment by xyzz** [comment]
It is not possible to achieve with the LCTL_T/etc functions as they can only support basic QMK keycodes: https://github.com/qmk/qmk_firmware/blob/master/docs/mod_tap.md#caveats
